src/app/issues-viewer/issues-viewer.component.ts

```diff
@@ -120,4 +124,21 @@ export class IssuesViewerComponent implements OnInit, AfterViewInit, OnDestroy {
       () => {}
     );
   }
+
+  /**
+   * Checks if our current repository available on phase service is indeed a valid repository
+   */
+  private checkIfValidRepository() {
+    const currentRepo = this.phaseService.currentRepo;
+
+    if (Repo.isEmptyRepo(currentRepo)) {
+      throw new Error('Invalid repo name. Please provide repo name in the format Org/Repo.');
+    }
+
+    this.phaseService.isValidRepository(currentRepo).subscribe((isValidRepository) => {
```

Comment:
Is there a reason for calling the ```isValidRepository``` method in ```phaseService``` rather than calling ```this.githubService.isRepositoryPresent(repo.owner, repo.name)``` directly?

Reply:
It was to keep it consistent with the rest of the changes in the other components. In confirm-login, it didn't have an existing dependency to githubService while some of the other components had it. I thought it was better to then use phase service instead.